bin/generate_related_references.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import subprocess
+
+import requests
+from github import Github
+from iteration_utilities import unique_everseen
+from ruamel.yaml import YAML
+
+GH_TOKEN = os.environ["GH_TOKEN"]
+GH_ORG_NAME = os.getenv("GH_ORG_NAME", "cloudposse")
+GH_SEARCH_PATTERN = os.getenv("GH_SEARCH_PATTERN", "terraform-")
+TF_MODULE_PATH = os.getenv("TF_MODULE_PATH", ".")
+TF_CONFIG_INSPECT_BINARY_PATH = os.getenv(
+    "TF_CONFIG_INSPECT_BINARY_PATH", "terraform-config-inspect"
+)
+TF_REGISTRY_URL = "https://registry.terraform.io/v1"
+
+gh = Github(GH_TOKEN)
+yaml = YAML(typ="rt")
+yaml.default_flow_style = False
+yaml.preserve_quotes = False
+
+
+def parse_gh():
+    gh_repos = []
+    for repo in gh.get_organization(GH_ORG_NAME).get_repos():
+        if GH_SEARCH_PATTERN in repo.name:
+            repo_object = {}
+            repo_object["name"] = repo.name
+            repo_object["description"] = repo.description
+            repo_object["url"] = repo.html_url
+            gh_repos.append(repo_object)
+    return gh_repos
+
+
+def tf_config_inspect():
+    output = json.loads(
+        subprocess.check_output(
+            [TF_CONFIG_INSPECT_BINARY_PATH, TF_MODULE_PATH, "--json"],
+            stderr=subprocess.STDOUT,
+        )
+    )
+    return output
+
+
+def parse_tf_registry(src_data, src_type):
+    items = []
+    src_item = "module_calls"
+    if src_type == "providers":
+        src_item = "required_providers"
+
+    for k, v in src_data[src_item].items():
+        item_object = {}
+        url = TF_REGISTRY_URL + "/" + src_type + "/" + v["source"]
+        r = requests.get(url=url).json()
+
+        if src_type == "providers":
+            name_pattern = "terraform-provider-{}".format(r["name"])
+        else:
+            name_pattern = "terraform-{}-{}".format(r["provider"], r["name"])
+        item_object["name"] = name_pattern
+        item_object["description"] = r["description"]
+        item_object["url"] = r["source"]
+        items.append(item_object)
+    return items
+
+
+if __name__ == "__main__":
+    related_reference_list = []
```

Comment:
So `related` and `references` are two separate sections of the README.

The `related` is for modules relates to this one.

The `references` section is for links to documentation (links to providers, link to AWS docs, etc) to read more, for example.